lib/ex_doc/language/erlang.ex

```diff
@@ -215,116 +260,140 @@ defmodule ExDoc.Language.Erlang do
     binary
   end
 
-  defp walk_doc({:code, _attrs, [{:a, _, _, _}], _meta} = ast, _config) do
+  defp walk_doc({:pre, _, _, _} = ast, _config) do
     ast
   end
 
-  defp walk_doc({:code, attrs, [code], meta} = ast, config) do
-    {text, url} =
-      case parse_autolink(code) do
-        {:module, module} = ref ->
-          text = Atom.to_string(module)
-          # Modules are parsed very permissively, so undefined/private
-          # modules do not emit warnings
-          url = do_url(ref, code, config, false)
-          {text, url}
-
-        {:local, kind, name, arity} ->
-          {"#{name}/#{arity}", local_url(kind, name, arity, config)}
-
-        {:remote, kind, module, name, arity} ->
-          text =
-            if kind == :type and arity == 0 do
-              "#{module}:#{name}()"
-            else
-              "#{module}:#{name}/#{arity}"
-            end
-
-          url = remote_url(kind, module, name, arity, config)
-          {text, url}
+  defp walk_doc({:code, attrs, [code], meta} = ast, config) when is_binary(code) do
+    ## Translate foo() to foo/0
+    link_code =
+      case Regex.run(~r{^(.+)\(\)$}, code) do
+        nil ->
+          code
 
-        :error ->
-          {code, nil}
+        [_, name] ->
+          name <> "/0"
       end
 
-    if url do
-      {:a, [href: url], {:code, attrs, [text], meta}, %{}}
-    else
-      ast
+    case Autolink.url(link_code, :regular_link, config) do
+      url when is_binary(url) ->
+        code = remove_prefix(code)
+        {:a, [href: url], [{:code, attrs, [code], meta}], %{}}
+
+      _ ->
+        ast
     end
   end
 
-  defp walk_doc({:a, attrs, inner, _meta} = ast, config) do
+  defp walk_doc({:a, attrs, inner, meta} = ast, config) do
     case attrs[:rel] do
       "https://erlang.org/doc/link/seeerl" ->
-        {fragment, url} = extract_fragment(attrs[:href] || "")
+        {fragment, url} = extract_fragment(attrs[:href] || "", "#")
 
         case String.split(url, ":") do
           [module] ->
-            autolink(:module, module, fragment, inner, config)
+            walk_doc({:a, [href: "`m:#{module}#{fragment}`"], inner, meta}, config)
 
           [app, module] ->
             inner = strip_app(inner, app)
-            autolink(:module, module, fragment, inner, config)
+            walk_doc({:a, [href: "`m:#{module}#{fragment}`"], inner, meta}, config)
 
           _ ->
             warn_ref(attrs[:href], config)
             inner
         end
 
       "https://erlang.org/doc/link/seemfa" ->
-        {kind, url} =
+        {prefix, url} =
           case String.split(attrs[:href], "Module:") do
-            [url] -> {:function, url}
-            [left, right] -> {:callback, left <> right}
+            [_] ->
+              case String.split(attrs[:href], "Mod:") do
```

Comment:
We can also do `String.split(..., ["Module:", "Mod:"])`. As you mentioned in the commit message perhaps we don't need it after all? I'd rather fix up the docs upstream. I have an old PR for diameter docs, https://github.com/erlang/otp/pull/5160, and happy to create more for stranglers if any.

Reply:
We can drop this commit, but the `Module` part needs to remain as edoc still uses the convention.

Reply:
So what uses "Mod:" and what uses "Module:"? We are very happy to support both but at the same time it feels like an opportunity to be consistent? Or is "Module:" edoc and "Mod:" the erlang docs?

Reply:
`Module` is edoc and most of Erlang/OTP. `Mod:` is only diameter as far as I know and that should be fixed.

I will remove the `Mod` parts from this commit.